engines/sci/engine/kernel_tables.h

```diff
@@ -407,25 +407,25 @@ static const SciKernelMapSubEntry kString_subops[] = {
-	{ SIG_SCI32,           9, MAP_DUMMY(ScrollWindowResize),       "o.",                   NULL },
-	{ SIG_SCI32,          10, MAP_CALL(ScrollWindowWhere),         "oi",                   NULL },
-	{ SIG_SCI32,          11, MAP_DUMMY(ScrollWindowGo),           "o..",                  NULL },
-	{ SIG_SCI32,          12, MAP_DUMMY(ScrollWindowInsert),       "o.....",               NULL },
-	{ SIG_SCI32,          13, MAP_DUMMY(ScrollWindowDelete),       "o.",                   NULL },
-	{ SIG_SCI32,          14, MAP_DUMMY(ScrollWindowModify),       "o.....(.)",            NULL },
-	{ SIG_SCI32,          15, MAP_DUMMY(ScrollWindowHide),         "o",                    NULL },
-	{ SIG_SCI32,          16, MAP_CALL(ScrollWindowShow),          "o",                    NULL },
-	{ SIG_SCI32,          17, MAP_CALL(ScrollWindowDestroy),       "o",                    NULL },
-	{ SIG_SCI32,          18, MAP_DUMMY(ScrollWindowText),         "o",                    NULL },
-	{ SIG_SCI32,          19, MAP_DUMMY(ScrollWindowReconstruct),  "o.",                   NULL },
+	{ SIG_SCI32,           1, MAP_CALL(ScrollWindowAdd),           "i.ii.(.)",             NULL },
+	{ SIG_SCI32,           2, MAP_DUMMY(ScrollWindowClear),        "i",                    NULL },
+	{ SIG_SCI32,           3, MAP_DUMMY(ScrollWindowPageUp),       "i",                    NULL },
+	{ SIG_SCI32,           4, MAP_DUMMY(ScrollWindowPageDown),     "i",                    NULL },
+	{ SIG_SCI32,           5, MAP_CALL(ScrollWindowUpArrow),       "i",                    NULL },
+	{ SIG_SCI32,           6, MAP_CALL(ScrollWindowDownArrow),     "i",                    NULL },
+	{ SIG_SCI32,           7, MAP_DUMMY(ScrollWindowHome),         "i",                    NULL },
+	{ SIG_SCI32,           8, MAP_DUMMY(ScrollWindowEnd),          "i",                    NULL },
+	{ SIG_SCI32,           9, MAP_DUMMY(ScrollWindowResize),       "i.",                   NULL },
+	{ SIG_SCI32,          10, MAP_CALL(ScrollWindowWhere),         "ii",                   NULL },
+	{ SIG_SCI32,          11, MAP_DUMMY(ScrollWindowGo),           "i..",                  NULL },
+	{ SIG_SCI32,          12, MAP_DUMMY(ScrollWindowInsert),       "i.....",               NULL },
+	{ SIG_SCI32,          13, MAP_DUMMY(ScrollWindowDelete),       "i.",                   NULL },
+	{ SIG_SCI32,          14, MAP_DUMMY(ScrollWindowModify),       "i.....(.)",            NULL },
+	{ SIG_SCI32,          15, MAP_CALL(ScrollWindowHide),          "i",                    NULL },
+	{ SIG_SCI32,          16, MAP_CALL(ScrollWindowShow),          "i",                    NULL },
+	{ SIG_SCI32,          17, MAP_CALL(ScrollWindowDestroy),       "i",                    NULL },
+	{ SIG_SCI32,          18, MAP_DUMMY(ScrollWindowText),         "i",                    NULL },
+	{ SIG_SCI32,          19, MAP_DUMMY(ScrollWindowReconstruct),  "i.",                   NULL },
```

Comment:
Have you checked LSL6? Is is using any other subops? I'm asking cause kScrollWindow is only used by these two games (SQ6 and LSL6 hires). If LSL6 is not using anything else, we can safely assume that the subcalls you've implemented are the only ones we need, and that kScrollWindow can be considered feature complete :)